internal/connector/connector.go

```diff
@@ -172,14 +169,8 @@ func Run(ctx context.Context, options Options) error {
 		return fmt.Errorf("parsing host config: %w", err)
 	}
 
-	clusterCACert, err := kubernetes.CA()
-	if err != nil {
-		return fmt.Errorf("reading CA file: %w", err)
-	}
-
 	certPool := x509.NewCertPool()
-
-	if ok := certPool.AppendCertsFromPEM(clusterCACert); !ok {
+	if ok := certPool.AppendCertsFromPEM(k8s.Config.CAData); !ok {
```

Comment:
The CA data is populated by the `rest.InClusterConfig` as long as we call `rest.LoadTLSFiles` (which I added to `NewKubernetes`). This removes the need to look it up again, and helps keep most the in-cluster logic in the `NewKubernetes` constructor.